### PR TITLE
adding sshproxy integration

### PR DESCRIPTION
**Pull request by wjcunningham7**
- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation, VERSION, and CHANGELOG accordingly.
- [ ] I have read the CONTRIBUTING document.

### CODE REVIEW

File: covalent_slurm_plugin/slurm.py [hunk 1]
```diff
@@ -136,6 +149,48 @@ async def _client_connect(self) -> Tuple[bool, asyncssh.SSHClientConnection]:
             The connection object
         """
 
+        if self.sshproxy:
+            try:
+                import oathtool
+            except ImportError:
+                raise RuntimeError(
+                    "To use 'sshproxy' options, reinstall the Slurm plugin as 'pip install covalent-slurm-plugin[sshproxy]'"
```
Comment (santoshkumarradha) [inline covalent_slurm_plugin/slurm.py:157]: Just nitpicking, I assume covalent needs to be restarted as well ?

Comment (wjcunningham7) [reply covalent_slurm_plugin/slurm.py:157]: I don't think so, but I can double check

Comment (wjcunningham7) [reply covalent_slurm_plugin/slurm.py:173]: confirmed server restart is not needed since it's imported at runtime

**Comment by arosen93** [comment]
https://github.com/AgnostiqHQ/covalent-slurm-plugin/blob/32a88d959d911aaac9102e91fd78e8a85112ab20/tests/slurm_test.py#L86 should be `executor.poll_freq == 60`. And if you wouldn't mind changing https://github.com/AgnostiqHQ/covalent-slurm-plugin/blob/32a88d959d911aaac9102e91fd78e8a85112ab20/tests/slurm_test.py#L100 to something > 60 like 90 that'd be great!

**Comment by arosen93** [comment]
Summariziing my comments from Slack:

1. I'd recommend linking to [here](https://docs.nersc.gov/connect/mfa/#creating-and-installing-a-token) in the instructions so that the user knows where to make the OTP.

2. I'd change "inspect the URL provided for a secret query string" to "inspect the provided URL for a string labeled secret="

3. For the line "In order to enable this behavior in this plugin, add the following block to your Covalent configuration while the server is stopped", I'd recommend hyperlinking "Covalent configuration" to the docs [here](https://covalent.readthedocs.io/en/latest/how_to/config/customization.html?highlight=configuration) in case the user hasn't done this before.

4.  Something to think about is that if a user opens an issue on GitHub and shares their `ct.get_config()`, they might inadvertently share the password and secret if they aren't knowledgeable or careful. This isn't restricted to the SLURM plugin, I should note.

5. The user has to have the sshproxy tool either in their PATH or in the current working directory. This could be easily overlooked and should be added to the docs.

6. It might be worthwhile briefly mentioning that the user can include the ssh key and certificate from sshproxy as input args if they want, i.e. they don't have to use the sshproxy interface [in case they are uncomfortable putting the info in their config.

7. The default for the "poll_freq" kwarg in SlurmExecutor() is 30 s (line 104) but should probably be raised to 60 s so you don't get the warning every time

8. The code is looking for `sshproxy`, but the file shipped by NERSC is `sshproxy.sh` (for `sshproxy.exe` for Windows). I had `sshproxy.sh` in my PATH and so the `sshproxy` command was not recognized.

**Pull request merged**